extensions-contrib/materialized-view-selection/src/main/java/io/druid/query/materializedview/MaterializedViewQueryRunner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.materializedview;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import io.druid.java.util.common.guava.MergeSequence;
+import io.druid.java.util.common.guava.Sequence;
+import io.druid.java.util.common.guava.Sequences;
+import io.druid.query.Query;
+import io.druid.query.QueryPlus;
+import io.druid.query.QueryRunner;
+
+import java.util.Map;
+
+
+public class MaterializedViewQueryRunner<T> implements QueryRunner<T>
+{
+  private final QueryRunner runner;
+  private final DataSourceOptimizer optimizer;
+  
+  public MaterializedViewQueryRunner(QueryRunner queryRunner, DataSourceOptimizer optimizer)
+  {
+    this.runner = queryRunner;
+    this.optimizer = optimizer;
+  }
+
+  @Override
+  public Sequence<T> run(QueryPlus<T> queryPlus, Map<String, Object> responseContext)
```

Comment:
I don't understand. Could you please describe it more detail? Thanks!

Reply:
Sure - not an important suggestion so please ignore if it seems irrelevant or too much work :)

In order to execute a materialised view query we have to issue multiple queries on different intervals and merge their results. That might be a more generally useful component where user's can union multiple queries rather than just multiple datasources.